src/makeCalculator.js

```diff
@@ -4,7 +4,47 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    operate(callback, value) {
+      this.result = callback.call(this, value);
+
+      return this;
+    },
+
+    add(value) {
+      this.result += value;
+
+      return this.result;
+    },
+
+    subtract(value) {
+      this.result -= value;
+
+      return this.result;
+    },
+
+    multiply(value) {
+      this.result *= value;
+
+      return this.result;
+    },
+
+    divide(value) {
+      this.result /= value;
+
+      return this.result;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+  };
+
+  return calculator;
```

Comment:
[NAMING] - The function is named `makeCalculator`, so the object returned should be named `calculator` to match the naming convention outlined in the checklist.